trestle/core/control_reader.py

```diff
@@ -799,48 +724,112 @@ def read_implemented_requirement(
         return sort_id, imp_req
 
     @staticmethod
-    def read_new_alters_and_params(control_path: pathlib.Path,
-                                   required_sections_list: List[str]) -> Tuple[str, List[prof.Alter], Dict[str, Any]]:
+    def read_new_alters_and_params(
```

Comment:
This function appears to be around 100 lines of the complex code, could this be simplified/separated?

Reply:
broke it up and simplified so it is now 50 lines.